docs/source/user_guide/archive_cache.rst

```diff
@@ -0,0 +1,109 @@
+===========================
+Using :mod:`.archive_cache`
+===========================
+
+:mod:`.archive_cache` provides three central fixtures 
+
+- :py:func:`~aiida_testing.archive_cache.create_node_archive` and :py:func:`~aiida_testing.archive_cache.load_node_archive` are used to manage AiiDA archive test files in the pytest environment
+- :py:func:`~aiida_testing.archive_cache.enable_archive_cache` uses the above fixtures and the AiiDA caching mechanisms to enable end-to-end tests of high-level workchains without going through executing individual `Calcjob` processes
+
+Here an example for using the :py:func:`~aiida_testing.archive_cache.enable_archive_cache` to test a simple workchain using the ``diff`` code
+
+The workchain looks as follows
+
+.. code-block:: python
+
+    class DiffWorkChain(WorkChain):
+    """
+    Very simple workchain which wraps a diff calculation for testing purposes
+    """
+
+    @classmethod
+    def define(cls, spec):
+        super(DiffWorkChain, cls).define(spec)
+        spec.expose_inputs(DiffCalculation, namespace='diff')
+        spec.outline(
+            cls.rundiff,
+            cls.results,
+        )
+        spec.output('computed_diff')
+
+    def rundiff(self):
+        inputs = self.exposed_inputs(DiffCalculation, 'diff')
+        running = self.submit(DiffCalculation, **inputs)
+
+        return ToContext(diff_calc=running)
+
+    def results(self):
+        computed_diff = self.ctx.diff_calc.get_outgoing().get_node_by_label('diff')
+        self.out('computed_diff', computed_diff)
+
+
+And an example of using the above fixtures in a pytest test suite looks as follows
+
+.. code-block:: python
+
+    def test_diff_workchain(enable_archive_cache):
+        """
+        End to end test of DiffWorkchain
+        """
+        # ... set up test inputs
+
+        inputs = {
+            'diff': {
+                'code': diff_code,
+                'parameters': parameters,
+                'file1': file1,
+                'file2': file2,
+            }
+        }
+    with enable_archive_cache('diff_workchain.aiida'):
+        res, node = run_get_node(DiffWorkChain, **inputs)
+
+    # Test results of workchain
+
+The fixture will look for the AiiDA archive named ``diff_workchain.aiida`` in a folder named ``caches`` in the same directory as the test file, if nothing else is specified.
+If this exists the archive is imported and the AiiDA caching functionality is enabled. All calculations created inside the with block will use the cached nodes if their
+inputs and attributes match.
+If the archive does not exist the workchain will try to run the complete calculation.
+
+.. note::
+    The caching mechanism of AiiDA is modified within test functions using these fixtures to ignore attributes, that would break the
+    caching if the tests are run on different machines with different versions of AiiDA for example
+    By default the stored test caches are also migrated to match the installed AiiDA version
```

Comment:
```suggestion
    The hashing mechanism of AiiDA is modified within tests that use the fixture to ignore certain attributes that would invalidate the
    cache when running the tests on different machines, with different versions of AiiDA, etc.
    By default, the test archives are migrated to match the installed AiiDA version.
```

Reply:
I guess this means you are ignoring the Computer on which the Code is set up.
Are you including any information about the Code in the hash?

Reply:
Only the name of the calculation plugin is included in the hash of the code